command/certificate/format.go

```diff
@@ -51,12 +64,72 @@ Convert PEM format to DER and write to disk:
 '''
 $ step certificate format foo.pem --out foo.der
 '''
+
+Convert a .p12 file to a certificate and private key:
+
+'''
+$ step certificate format foo.p12 --crt foo.crt --key foo.key --format pem
+'''
+
+Convert a .p12 file to a certificate, private key and intermediate certificates:
+
+'''
+$ step certificate format foo.p12 --crt foo.crt --key foo.key --ca intermediate.crt --format pem
+'''
+
+Convert a certificate and private key to a .p12 file:
+
+'''
+$ step certificate format foo.crt --crt foo.p12 --key foo.key --format p12
+'''
+
+Convert a certificate, a private key, and intermediate certificates(s) to a .p12 file:
+
+'''
+$ step certificate format foo.crt --crt foo.p12 --key foo.key \
+  --ca intermediate-1.crt --ca intermediate-2 --format p12
+'''
 `,
 		Flags: []cli.Flag{
 			cli.StringFlag{
-				Name:  "out",
-				Usage: `Path to write the reformatted result.`,
+				Name: "format",
+				Usage: `The desired output <format> for the input. The default behavior is to
+convert between DER and PEM format. Acceptable inputs are 'pem', 'der', and 'p12'.`,
+			},
+			cli.StringFlag{
+				Name: "crt",
+				Usage: `The path to a certificate <file>. If --format is 'p12' then this flag
+must be a PEM or DER encoded certificate. If the positional argument is a P12
+encoded file then this flag contains the name for the PEM or DER encoded leaf
+certificate extracted from the p12 file.`,
 			},
+			cli.StringFlag{
+				Name: "key",
+				Usage: `The path to a key <file>. If --format is 'p12' then this flag
+must be a PEM or DER encoded private key. If the positional argument is a P12
+encoded file then this flag contains the name for the PEM or DER encoded private
+key extracted from the p12 file.`,
+			},
+			cli.StringSliceFlag{
+				Name: "ca",
+				Usage: `The path to a root or intermediate certificate <file>. If --format is 'p12'
+then this flag can be used to submit one or more CA files encoded as PEM or DER.
+Additional CA certificates can be added by using the --ca flag multiple times.
+If the positional argument is a p12 encoded file then this flag contains the
+name for the PEM or DER encoded certificate chain extracted from the p12 file.`,
```

Comment:
It's pretty easy to offer this utility and just explain how the intermediates are ordered. Since it's already implemented this way and not particularly complicated, I'm not sure if switching back makes sense. Maybe just adding some better documentation.